### PR TITLE
[Ready] Update json format as defined in JSON API v1

### PR DESCRIPTION
**Pull request by GlennGeelen**
http://jsonapi.org/format/

- [x] rename links to relationships
- [x] rename linked to included
- [x] add a data object or array to the relationships
- [x] put attributes of the object type in an attributes object
- [ ] make links work like defined in the documentation
- [x] make related links work like in the documentation
- [x] remove SerializeAsOption.Embedded (this should never be possible)
- [ ] refactor long functions to smaller, more understandable functions
- [x] update unit tests (all test data is updated to what the output should be)

### CODE REVIEW

File: JSONAPI/Json/JsonApiFormatter.cs [hunk 1]
```diff
@@ -185,91 +185,32 @@ protected void Serialize(object value, Stream writeStream, JsonWriter writer, Js
...
-                }
-            }
+            // Write attributes
+            WriteAttributes(props, writer, idProp, value, serializer, relationshipModelProperties);
 
             // Now do other stuff
             if (relationshipModelProperties.Count() > 0)
             {
-                writer.WritePropertyName("links");
+                writer.WritePropertyName("relastionships");
```
Comment (csantero) [inline JSONAPI/Json/JsonApiFormatter.cs]: `relationships`

**Comment by csantero** [comment]
Thanks! I've been meaning to get to this for a while now but just haven't find the time.

File: JSONAPI/Json/JsonApiFormatter.cs (hunk 1 above)
Comment (GlennGeelen) [inline JSONAPI/Json/JsonApiFormatter.cs]: thanks, i was wandering why it wasn't working properly :)

**Comment by GlennGeelen** [comment]
Everything for output serializing is fixed like described in JSON API v1 except for Links. For now the SerializeAsLink with a linktemplate will serialize like this:
```
"post": {
  "links": {
    "related": "http://example.com/posts/1"
  }
}
```
where "http://example.com/posts/1" is the linktemplate.

Next week I'll focus on getting the links done the right way, refactor some more and fix the tests.
I'll try to look into the relationships of a POST and PUT.

**Comment by csantero** [comment]
Looking good, thanks!

**Comment by GlennGeelen** [comment]
So far I only implemented the related links option. This actually the only kind of links we want to use but there are options for the self link and both links as well. 

Futher more I fixed all the tests except for the filter options. 

This is as far as I can help you right now, I don't know when I have more time to do the other things. So, could you take a look at it @csantero

**Comment by csantero** [comment]
@GlennGeelen I've added a commit fixing all the remaining tests. There's still a ways to go before we can call ourselves fully 1.0-compatible, but this was a big step in the right direction. Thanks for all your hard work!

**Comment by GlennGeelen** [comment]
@csantero thanks for fixing the remaining tests and merging.

**Pull request merged**